lib/extras.py

```diff
@@ -437,6 +438,150 @@ def callproc(self, procname, vars=None):
         return LoggingCursor.callproc(self, procname, vars)
 
 
+class ReplicationConnection(_connection):
+    """A connection that uses `ReplicationCursor` automatically."""
+
+    def __init__(self, *args, **kwargs):
+        """Initializes a replication connection, by adding appropriate replication parameter to the provided dsn arguments."""
+
+        if len(args):
+            dsn = args[0]
+
+            # FIXME: could really use parse_dsn here
+
+            if dsn.startswith('postgres://') or dsn.startswith('postgresql://'):
+                # poor man's url parsing
+                if dsn.rfind('?') > 0:
+                    if not dsn.endswith('?'):
+                        dsn += '&'
+                else:
+                    dsn += '?'
+            else:
+                dsn += ' '
+            dsn += 'replication=database'
```

Comment:
replication=database won't work on 9.3 or older, fwiw... but it's useful to have replication=true supported for 9.4+ too, because that lets you connect without knowing a particular database name when you plan to do streaming replication.

I suggest appending `replication=database` only if you don't already find a `replication` dsn term. That way the user can explicitly pass `replication=on` if they really want it, and it doesn't require any complicated API that's more effort than it's worth to add.